components/script/dom/htmlareaelement.rs

```diff
@@ -12,10 +12,130 @@ use dom::domtokenlist::DOMTokenList;
 use dom::htmlelement::HTMLElement;
 use dom::node::Node;
 use dom::virtualmethods::VirtualMethods;
+use euclid::point::Point2D;
 use std::default::Default;
 use string_cache::Atom;
 use style::attr::AttrValue;
 
+pub enum Area {
+    Circle { left: f32, top: f32, radius: f32 },
+    Rectangle { left_l: f32, top_t: f32, left_r: f32, top_b: f32 },
+    Polygon { points: Vec<f32> },
+    Default,
+}
+
+// https://html.spec.whatwg.org/multipage/#rules-for-parsing-a-list-of-floating-point-numbers
+impl Area {
+    pub fn get_area(coord: &str) -> Area {
+        let mut array;
+        let size = coord.len();
+        let num = coord.as_bytes();
+        let mut index = 0;
+        let mut number_list = Vec::new();
+
+        // Step 4: Walk till char is not a delimiter
+        while index < size
+        {
+            let val = num[index];
+            match val {
+                0x2C | 0x3B | 0x20 | 0x09 | 0x0A | 0x0C | 0x0D  => {},
+                _ => break,
+            }
+
+            index += 1;
+        }
+
+        // Step 5: walk till end of string
+        while index < size
+        {
+            array = Vec::new();
+
+            // Step 5.1: walk till we hit a valid char i.e., 0 to 9, dot or dash, e, E
+            while index < size
+            {
+                let val = num[index];
+                match val {
+                    0x30...0x39 | 0x2E | 0x2D | 0x45 | 0x65 => break,
+                    _ => {},
+                }
+
+                index += 1;
+            }
+
+            // Step 5.2: collect valid symbols till we hit another delimiter
+            while index < size
+            {
+                let val = num[index];
+
+                match val {
+                    0x30...0x39 | 0x2E | 0x2D | 0x45 | 0x65 => array.push(num[index]),
+                    _ => break,
+                }
+
+                index += 1;
+            }
+
+            // The input does not consist any valid charecters
+            if array.is_empty() {
+                continue;
+            }
+
+            // Convert String to float
+            let mut string_val = String::from_utf8(array);
+            match string_val.unwrap().parse::<f32>() {
+                Ok(v) => number_list.push(v),
+                Err(_) => number_list.push(0.0),
+            };
+        }
+
+        let final_size = number_list.len();
+
+        if final_size == 3 {
```

Comment:
In [HTMLAreaElement.idl](https://github.com/servo/servo/blob/master/components/script/dom/webidls/HTMLAreaElement.webidl), we noticed that all the attributes are commented out. We were thinking if these are necessary to access 'area' elements attributes such as shape and coords. We were not sure how to access these attributes. 
Also, we were thinking of seeing which of the values are populated by running a debugger on servo, but we were also not sure of which of the HTML features are implemented on Servo.